utils/docker/Dockerfile.centos.7

```diff
@@ -33,13 +38,30 @@ baseurl=${MY_REPO}\n\
 enabled=1\n\
 repo_gpgcheck=0\n\
 exclude=fuse-3* fuse-devel-3* fuse-libs-3* fuse-debuginfo-3* ompi*\n\
-gpgcheck=0\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
-      yum clean all;                                            \
-      yum-config-manager --assumeyes --quiet                    \
+gpgcheck=0\n" >> /etc/yum.repos.d/local-daos-group.repo;        \
+      dnf clean all;                                            \
+      dnf config-manager --assumeyes --quiet                    \
+                         --disable base epel extra updates;     \
+    fi
+
+# If a local distro repository is supplied, then we should use it
+# and disable using the distro repositories.
+RUN if [ -n "$REPO_DISTRO" ]; then                              \
+      MY_REPO="${REPO_URL}${REPO_DISTRO}/";                     \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
+      echo -e "[${MY_NAME}]\n\
+name=created from ${MY_REPO}\n\
+baseurl=${MY_REPO}\n\
+enabled=1\n\
+repo_gpgcheck=0\n\
+gpgcheck=1\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
+      dnf clean all;                                            \
+      dnf config-manager --assumeyes --quiet                    \
                          --disable base epel extra updates;     \
-    fi;                                                         \
-    sed "s/^mirrorlist_expire=0*/mirrorlist_expire=99999999/"   \
-        -i /etc/yum.conf
+      sed "s/^mirrorlist_expire=0*/mirrorlist_expire=99999999/" \
+          -i /etc/yum.conf;                                     \
+    fi
```

Comment:
Would it be better to move this to the very top of the file and do it all with `yum` so as to avoid using upstream repos for even this step?  In any case, this step seems more stable/consistent across runs so it should be higher in the file to better take advantage of caching and not be cache-busted if/when the quickbuild repo add is being done.

Reply:
There is some dependency issues that force this order.

The epel-release must be installed and then updated before installing the local mirror in order to get the current GPG key for it.

Reply:
I have adjusted it as much as I think that I can for this pass.

The epel-release must be installed and updated before Yum/DNF will even use a Nexus group repo that contains the EPEL repo.

So moved the cachebusting behind having the EPEL repository installed, which should make our exposure to external mirrors smaller.